### PR TITLE
Fix pipeline round-trip serialization

### PR DESCRIPTION
**Pull request by mhamilton723**
- Made the pipeline stage param a new param type
- Fix a problem with the complexTypeConverter
- fix the spark hardcoded namespace problem

### CODE REVIEW

File: src/utils/src/main/python/java_params_patch.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    Meta-algorithms such as Pipeline should override this method as a classmethod.
+    """
+    def __get_class(clazz):
+        """
+        Loads Python class from its name.
+        """
+        parts = clazz.split('.')
+        module = ".".join(parts[:-1])
+        m = __import__(module)
```
Comment (drdarshan) [inline src/utils/src/main/python/java_params_patch.py]: Does this work in Python3? I thought you need to use imporlib?

**Comment by drdarshan** [comment]
I don't see examples of how this is used? It looks like you need to implement a special params subtype for this logic to kick in?

File: src/utils/src/main/python/java_params_patch.py (hunk 1 above)
Comment (mhamilton723) [inline src/utils/src/main/python/java_params_patch.py]: this is verbatim from spark, just with an additional line. I tested it in python3

**Comment by mhamilton723** [comment]
I will have some stuff in the next set of PRs that uses the pipeline stage param. I figured it would be useful to have in general, if you would like it taken out until then  I can do that

File: src/utils/src/main/python/java_params_patch.py (hunk 1 above)
Comment (drdarshan) [inline src/utils/src/main/python/java_params_patch.py]: Nice!

**Pull request merged**